modulus/utils/geometric/convert_file_formats.py

```diff
@@ -0,0 +1,98 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+
+import vtk
+
+
+def convert_obj_to_vtp(input_file: str, output_file: str) -> None:
+    """
+    Convert an OBJ file to a VTP file.
+
+    Args:
+    - input_file (str): Path to the input OBJ file.
+    - output_file (str): Path to save the converted VTP file.
+    """
+    reader = vtk.vtkOBJReader()
+    reader.SetFileName(input_file)
+    if not reader.CanReadFile(input_file):
+        raise ValueError(f"Error: Could not read file: {input_file}")
+    reader.Update()
+
+    polydata = reader.GetOutput()
+
+    writer = vtk.vtkXMLPolyDataWriter()
+    writer.SetFileName(output_file)
+    writer.SetInputData(polydata)
+    writer.Write()
+
+
+def convert_vtp_to_stl(input_file: str, output_file: str) -> None:
+    """
+    Convert a VTP file to an STL file.
+
+    Args:
+    - input_file (str): Path to the input VTP file.
+    - output_file (str): Path to save the converted STL file.
+    """
+    reader = vtk.vtkXMLPolyDataReader()
+    reader.SetFileName(input_file)
+    if not reader.CanReadFile(input_file):
+        raise ValueError(f"Error: Could not read file: {input_file}")
+    reader.Update()
```

Comment:
Don't you have to extract surfaces alone? Or alteast check that the loaded vtp is a surface data? 

I thought something like this was required:

```
    # Convert unstructured grid to polydata.
    surface_filter = vtk.vtkDataSetSurfaceFilter()
    surface_filter.SetInputConnection(reader.GetOutputPort())
    surface_filter.Update()
```

Reply:
Clarified the scope